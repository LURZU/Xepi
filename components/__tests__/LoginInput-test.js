import React from 'react';
import renderer from 'react-test-renderer';
import { useContext } from 'react';
import LoginInform  from '../../components/provider/LogInForm';
// import { AuthContext } from '../../components/provider/AuthContext';
import { render, fireEvent, findAllByType  } from '@testing-library/react-native';
// import {create, TestRenderer }from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<LoginInform />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test('Password input check', () => {
    
//       const { getByTestId } = render(<LoginInform />);
//       const contextProvider = getByTestId('auth-password');
//       fireEvent.changeText(contextProvider, { target: { value: 'test password' } });
//       expect(contextProvider.value).toEqual('test password'); 
     
//   });


  test('Password auth check', () => {
    const { getByTestId } = render(<LoginInform />);
    const input = getByTestId('auth-password');
    fireEvent.changeText(input, '123');
    expect(input.props.value).toBe('123');
  });

  
  test('Login auth check', () => {
    const { getByTestId } = render(<LoginInform />);
    const input = getByTestId('auth-login');
    fireEvent.changeText(input, '123');
    expect(input.props.value).toBe('123');
  });

  test('Button auth check', () => {
    const { getByTestId } = render(<LoginInform />);
    const input = getByTestId('auth-button');
    fireEvent.press(input);
  });


  test('Context ', async () => {
    const { getByTestId } = render(
      <LoginInform />
    );
    const inputLog = getByTestId('auth-login'); 
    fireEvent.changeText(inputLog, 'test');
    const inputPass = getByTestId('auth-password'); 
    fireEvent.changeText(inputPass, 'test');
    const ModalAuth = getByTestId('auth-button');
    fireEvent.press(ModalAuth);
  
    const modalVisibility = getByTestId('auth-modal');
    const allModals = await findAllByType(modalVisibility.type);
  
    expect(allModals[1].props.visible).toBe(true);
  });