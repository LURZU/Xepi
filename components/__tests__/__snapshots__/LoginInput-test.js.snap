// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="auth-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "color": "#FFFFFF",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 30,
          "fontWeight": "bold",
          "left": 30,
          "marginBottom": 30,
          "position": "absolute",
          "textAlign": "left",
          "top": 30,
        }
      }
    >
      Connectez-vous
    </Text>
    <View
      style={
        {
          "borderColor": "#FFFFFF",
          "borderRadius": 20,
          "borderWidth": 1,
          "color": "#FFFFFF",
          "flexDirection": "row",
          "height": 50,
          "marginBottom": 15,
          "paddingHorizontal": 20,
          "width": "100%",
        }
      }
    >
      <Text />
      <TextInput
        onChangeText={[Function]}
        placeholder="Nom d'utilisateur"
        placeholderTextColor="#FFF"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000",
            "color": "#FFFFFF",
            "flex": 1,
            "height": 47,
            "justifyContent": "flex-start",
          }
        }
        testID="auth-login"
        value=""
      />
    </View>
    <View
      style={
        {
          "borderColor": "#FFFFFF",
          "borderRadius": 20,
          "borderWidth": 1,
          "color": "#FFFFFF",
          "flexDirection": "row",
          "height": 50,
          "marginBottom": 15,
          "paddingHorizontal": 20,
          "width": "100%",
        }
      }
    >
      <Text />
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#FFF"
        secureTextEntry={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000",
            "color": "#FFFFFF",
            "flex": 1,
            "height": 47,
            "justifyContent": "flex-start",
          }
        }
        testID="auth-password"
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "margin": "auto",
            "marginLeft": "auto",
            "position": "relative",
            "right": 0,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "color": "#FFFFFF",
          "width": "100%",
        }
      }
      testID="auth-button"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4292159654,
              4287513773,
            ],
            "endPoint": [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          {
            "borderRadius": 15,
            "marginTop": 50,
            "paddingHorizontal": 40,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "textAlign": "center",
            }
          }
        >
          Se connecter
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</Modal>
`;
